src/metabase/legacy_mbql/schema.cljc

```diff
@@ -912,12 +912,16 @@
                        (is-clause? numeric-functions x)  :numeric
                        (is-clause? string-functions x)   :string
                        (is-clause? boolean-functions x)  :boolean
+                       (is-clause? :value x)             :value
+                       (is-clause? :segment x)           :segment
                        :else                             :else))}
    [:datetime DatetimeExpression]
    [:numeric  NumericExpression]
    [:string   StringExpression]
    [:boolean  BooleanExpression]
-   [:else     (one-of segment)]])
+   [:value    value]
+   [:segment  segment]
+   [:else     Field]])
```

Comment:
Don't we also want to allow `:expression` references here? Or maybe update the `BooleanExpression` schema to allow them?

Reply:
`Field` includes both `:field` and `:expression` :)

Reply:
TIL!